numba/tests/test_array_reductions.py

```diff
@@ -938,6 +938,68 @@ def a_variations():
             for a in a_variations():
                 check(a)
 
+    def test_argmax_axis_1d_2d_4d(self):
+        arr1d = np.array([0, 20, 3, 4])
+        arr2d = np.arange(6).reshape(2, 3)
+        arr2d[0,1] += 100
+
+        arr4d = np.arange(120).reshape(2, 3, 4, 5) + 10
+        arr4d[0,1,1,2] += 100
+        arr4d[1,0,0,0] += 100
```

Comment:
```suggestion
        arr4d[0, 1, 1, 2] += 100
        arr4d[1, 0, 0, 0] -= 51
```
to add a bit more variation